components/tidb_query/src/expr_util/conv.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::codec::data_type::*;
+
+#[derive(Copy, Clone)]
+struct IntWithSign(u64, bool);
+
+impl IntWithSign {
+    fn from_int(num: Int) -> IntWithSign {
+        IntWithSign(num.wrapping_abs() as u64, num < 0)
+    }
+
+    fn from_signed_uint(num: u64, is_neg: bool) -> IntWithSign {
+        IntWithSign(num, is_neg)
+    }
+
+    // Shrink NUM to fit the boundary of i64.
+    fn shrink_from_signed_uint(num: u64, is_neg: bool) -> IntWithSign {
+        let value = if is_neg {
+            num.min(-Int::min_value() as u64)
+        } else {
+            num.min(Int::max_value() as u64)
+        };
+        IntWithSign::from_signed_uint(value, is_neg)
+    }
+
+    fn format_radix(mut x: u64, radix: u32) -> String {
```

Comment:
would you mind adding `assert(is_valid_base(radix))` here?

Reply:
What about `debug_assert!` ?

Reply:
ping @mapleFU

Reply:
pong

Reply:
Yes, `debug_assert` is ok here.

Reply:
I'm sorry, I want to commit this but it was merged